Function/Books.cs

```diff
@@ -7,64 +7,155 @@
+        await response.WriteAsJsonAsync(BookModel.FromBookTableModel(queryResult.Value));
+        return response;
+    }
+
+    // create new book
+    [OpenApiOperation(operationId: "createBook", tags: new[] {"books"}, Summary = "Create Book", Description = "Create a new book in the backend.")]
+    [OpenApiSecurity("function_key", SecuritySchemeType.ApiKey, Name = "code", In = OpenApiSecurityLocationType.Query)]
+    [OpenApiRequestBody(contentType: "application/json", bodyType: typeof(BookModel))]
+    [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json", bodyType: typeof(BookModel))]
+    [OpenApiResponseWithBody(statusCode: HttpStatusCode.BadRequest, contentType: "application/json", bodyType: typeof(ErrorModel))]
+    [Function("HTTPCreateBook")]
+    public async Task<HttpResponseData> CreateBook([HttpTrigger(AuthorizationLevel.Function, "post", Route = "book")] HttpRequestData request)
+    {
+        _logger.LogInformation($"[{request.FunctionContext.InvocationId}] Processing request for create book endpoint.");
+
+        // deserialize request body into BookModel object
+        var createBookReq = await request.ReadFromJsonAsync<BookModel>();
+
+        // if request body cannot be deserialized or is null, return an HTTP 400
+        if (createBookReq == null)
+            return request.CreateResponse(HttpStatusCode.BadRequest);
+
+        // if isbn from book create request already exists -> return an HTTP 400
+        if (_table.GetEntityIfExists<BookTableModel>(partitionKey: string.Empty, rowKey: createBookReq.Isbn).HasValue)
+            return request.CreateResponse(HttpStatusCode.BadRequest);
+
+        // transform BookModel into BookTableModel and write row to table; partition + row key need to be unique!
+        var createTableRow = await _table.AddEntityAsync<BookTableModel>(new()
         {
-            _logger.LogWarning($"[{request.FunctionContext.InvocationId}] Error: {bookTitle} was not found.");
-            return request.CreateResponse(HttpStatusCode.NotFound);
+            RowKey = createBookReq.Isbn,
+            Title = createBookReq.Title,
+            Author = createBookReq.Author,
+            PublishYear = createBookReq.PublishYear
+
+        });
 
+        // return error if transaction in table storage unsuccessfull
+        if (createTableRow.IsError)
+        {
+            var errorResponse = request.CreateResponse(HttpStatusCode.InternalServerError);
+            await errorResponse.WriteAsJsonAsync<ErrorModel>(new
+            (
+                Error: "TableTransactionError",
+                ErrorMessage: "There was a problem executing the table transaction."
+            ));
+            return errorResponse;
         }
+
+        // serialize requested BookModel to json and return to client, when request successfull
+        var response = request.CreateResponse(HttpStatusCode.OK);
+        await response.WriteAsJsonAsync(createBookReq);

```

Comment:
It is common in a POST HTTP method that creates an object / item to return the created item, when successful. That is what we are doing here.